src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -87,45 +87,47 @@ internal object Deserialize(BinaryParser serParser, bool fCheck)
 
             _isSimpleAssembly = (_formatterEnums._assemblyFormat == FormatterAssemblyStyle.Simple);
 
-
-            if (_fullDeserialization)
+            using (DeserializationToken token = SerializationInfo.StartDeserialization())
```

Comment:
Does this cast the struct to an `IDisposable` and box it? If so, consider changing this to a try/finally.

Reply:
> Does this cast the struct to an IDisposable and box it? If so, consider changing this to a try/finally.

Nah, the `using` statement is efficient for IDisposable structs (provided that the structness is exposed via the API, which this one is)